pynest/nest/tests/test_poisson_generator_rate_change.py

```diff
@@ -31,44 +31,56 @@ class TestPgRateChange(unittest.TestCase):
     def test_statistical_rate_change(self):
         """Statistical test of poisson_generator_ps rate change"""
 
-        # Lower limit of acceptable p_value
-        p_value_lim = 0.3
+        p_value_lim = 0.05  # Lower limit of acceptable p_value
+        resolution = 0.25  # Simulation resolution
+        n_parrots = 1000  # Number of parrot neurons
+        sim_time = 100  # Time to simulate
 
-        def kstest_first_spiketimes(sd, start_t):
+        def kstest_first_spiketimes(sd, start_t, rate):
```

Comment:
Code in `ks_first_spiketimes()` depends in `resolution` and `n_parrots` defined in the surrounding scope. That is neither transparent nor safe. I would suggest to pass them as arguments along with `rate`. I'd also suggest to make the function a private method in `TestPgRateChange` instead of defining it as an internal function here.